packages/dev/core/src/Cameras/Inputs/arcRotateCameraMouseWheelInput.ts

```diff
@@ -80,19 +83,27 @@ export class ArcRotateCameraMouseWheelInput implements ICameraInput<ArcRotateCam
      * @param noPreventDefault Defines whether event caught by the controls should call preventdefault() (https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault)
      */
     public attachControl(noPreventDefault?: boolean): void {
+        const deviceSourceManager = this.camera._deviceSourceManager;
+        // If the user tries to attach this control without having general camera controls active, warn and return.
+        if (!deviceSourceManager) {
+            Logger.Warn("Cannot attach control to camera.  Camera controls not present");
+            return;
+        }
+
         noPreventDefault = Tools.BackCompatCameraNoPreventDefault(arguments);
 
-        this._connectedObserver = this.camera._deviceSourceManager!.onDeviceConnectedObservable.add((deviceSource) => {
+        this._connectedObserver = deviceSourceManager.onDeviceConnectedObservable.add((deviceSource) => {
             if (deviceSource.deviceType === DeviceType.Mouse) {
                 this._observer = deviceSource.onInputChangedObservable.add((eventData) => {
-                    if ("deltaY" in eventData) {
-                        this._wheel(eventData);
+                    const type = GestureRecognizer.DeterminePointerEventType(deviceSource, eventData);
+                    if (type & PointerEventTypes.POINTERWHEEL) {

```

Comment:
nit: Style is inconsistent with the rest of the code which uses `(a & b) !== 0`. For size savings, this style is probably better.